app/scripts/lib/rpc-method-middleware/handlers/ethereum-chain-utils.ts

```diff
@@ -186,20 +217,27 @@ export function validateAddEthereumChainParams(params, end) {
 }
 
 export async function switchChain(
-  res,
-  end,
-  origin,
-  chainId,
-  requestData,
-  networkClientId,
-  approvalFlowId,
+  res: JsonRpcRequest,
```

Comment:
Looks like this is a typo?

```suggestion
  res: JsonRpcResponse<Json>,
```

or 

```suggestion
export async function switchChain<Result extends Json = Json>(
  res: JsonRpcResponse<Result>,
```

I'm seeing `JsonRpc{Request,Response}` resolve to implicit `any` in this file. I'll see if I can find out more about this.